src/validation.cpp

```diff
@@ -1394,6 +1394,11 @@ PackageMempoolAcceptResult MemPoolAccept::AcceptPackage(const Package& package,
             // Provide the wtxid of the mempool tx so that the caller can look it up in the mempool.
             results_final.emplace(wtxid, MempoolAcceptResult::MempoolTxDifferentWitness(iter.value()->GetTx().GetWitnessHash()));
         } else {
+            if (wtxid == child->GetWitnessHash() && !quit_early) {
+                txns_package_eval.push_back(tx);
+                // Unless we're quitting early, validate the child outside of this loop.
```

Comment:
cb065422e58be7558e6c2ad9eadfa885bb48c708

What you're saying in the commit message is true only for `TxValidationResult::TX_MEMPOOL_POLICY` or `TxValidationResult::TX_MISSING_INPUTS`, because in that case we won't re-validate the child, right? So it's only the other failures that trigger validating twice.

What's even worse than a mistake in the commit message, you won't be able to catch other kinds of failures here (because you don't set `quit_early` here), so you might do some extra work which should have been avoided otherwise, no?

Reply:
Added in the commit message that, specifically if the failure is policy or missing inputs, it was validated twice. Thanks